### PR TITLE
Subfunction Fix

### PR DESCRIPTION
**Pull request by abhishek-singh591**
This PR fixes subfunction-based export issues for the following models:

1. `bigcode/starcoder`  
2. `ibm-granite/granite-20b-code-base-8k`  
3. `ibm-granite/granite-20b-code-instruct-8k`  
4. `Qwen3-30B-A3B-Instruct-2507`  
5. `Mixtral-8x7B`

In addition, it updates the Causal LM subfunction test file to make it more robust and resilient across models.

### CODE REVIEW

**Review by quic-rishinr — APPROVED** [review]

File: QEfficient/transformers/models/granitemoe/modeling_granitemoe.py [hunk 1]
```diff
@@ -460,7 +460,7 @@ def forward(self, layer_input):
         final_hidden_states = torch.zeros_like(layer_input)
         for expert_idx in range(num_experts):
             mask = expert_mask[expert_idx].transpose(0, 1).to(layer_input.dtype)
-            mask_weight = (topk_gates * mask).sum(dim=1, keepdim=True)
+            mask_weight = torch.einsum("be,be->b", topk_gates, mask.to(topk_gates.dtype))[:, None]
```
Comment (vbaddi) [inline QEfficient/transformers/models/granitemoe/modeling_granitemoe.py]: nit: I don't this is needed, the `sum(dim=1)` should be okay. Are you seeing the issue w/this too?
